wallet-new/src/Cardano/Wallet/API/V1/ReifyWalletError.hs

```diff
@@ -0,0 +1,351 @@
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE TypeApplications    #-}
+
+module Cardano.Wallet.API.V1.ReifyWalletError (
+    translateKernelErrors
+    ) where
+
+import qualified Data.Text as T
+
+import           Formatting (build, sformat)
+import           Universum
+
+import           Pos.Core (decodeTextAddress)
+
+import qualified Cardano.Wallet.API.V1.Types as V1
+import           Cardano.Wallet.API.V1.Types (V1 (..))
+import qualified Cardano.Wallet.Kernel.Accounts as Kernel
+import qualified Cardano.Wallet.Kernel.Addresses as Kernel
+import           Cardano.Wallet.Kernel.CoinSelection.Generic (CoinSelHardErr (..))
+import           Cardano.Wallet.Kernel.DB.AcidState (NewPendingError (..),
+    NewForeignError (..))
+import qualified Cardano.Wallet.Kernel.DB.HdWallet as HD
+import qualified Cardano.Wallet.Kernel.DB.HdWallet.Create as HD
+import qualified Cardano.Wallet.Kernel.Transactions as Kernel
+import qualified Cardano.Wallet.Kernel.Wallets as Kernel
+import           Cardano.Wallet.WalletLayer
+import           Cardano.Wallet.WalletLayer.Kernel.Conv (toRootId)
+
+-- | Translates all kernel exceptions to the WalletError sum type and returns
+--   Nothing for non-kernel exceptions.
+translateKernelErrors :: SomeException -> Maybe V1.WalletError
+translateKernelErrors ex = do
```

Comment:
Minor wrinkle on names: I think in principle it would be better to rename this to `translateWalletLayerErrors` because we are not working directly with the kernel (that's low level) but we are interacting with the `WalletLayer`, which is the abstraction layer the `Application` sees directly 😉